lib/drivers/veres.js

```diff
@@ -768,15 +774,205 @@ api.notes = async options => {
+
+  // public ledger key
+  const newLedgerKey = {
+    id: nextKeyId,
+    owner: didDocument.id
+  };
+  if(options.type === 'ed25519') {
+    newLedgerKey.type = 'Ed25519VerificationKey2018';
+    if(options.public) {
+      newLedgerKey.publicKeyBase58 = options.public;
+    }
+  } else if(options.type === 'rsa') {
+    throw new Error('not implemented');
+  }
+  // clone and add to private local key
+  const newLocalKey = util.deepClone(newLedgerKey);
+  if(options.type === 'ed25519') {
+    if(options.private) {
+      newLocalKey.privateKey = {
+        privateKeyBase58: options.private
+      };
+    }
+  } else if(options.type === 'rsa') {
+    throw new Error('not implemented');
+  }
+  // add to local and remote docs
+  jsonld.addValue(params, 'publicKey', newLedgerKey);
+  jsonld.addValue(ledgerParams, 'publicKey', newLocalKey);
+  //console.log('LEDGER', JSON.stringify(ledgerDidDocument, null, 2));
+  //console.log('LOCAL', JSON.stringify(didDocument, null, 2));
+
+  // FIXME: reusing import code, rename it
+  _import(options, didDocument);
+
+  const operation = didv1.wrap(
+    {didDocument: ledgerDidDocument, operationType: 'update'});
+  //console.log(JSON.stringify(operation, null, 2));
+
+  await release();
+
+  _log(options, 'DID local update successful.');
+
+  //if(options.send) {
+    await _send(options, ledgerDidDocument, didDocument);
+  //}
 };
 
+function _removeKeyId(didDocument, id) {
+  // check all suites
+  for(const suiteId of _suiteIds) {
+    for(const suiteParams of jsonld.getValues(didDocument, suiteId)) {
+      suiteParams.publicKey = jsonld.getValues(suiteParams, 'publicKey')
+        .filter(key => key.id !== id);
+    }
+  }
+}
+
 api['authn-remove'] = async options => {
-  throw new Error('authn-remove not implemented');
+  _log(options, 'authn-add', options);
+  const did = options.did;
+  // lock local
+  const release = await _lockDid(did, options);
+  // get local did doc
+  const didDocument = await keyStorage.load(did);
+  // get remote did doc
+  const hostname = _getHostname(options);
+  const ledgerDidDocument = await didcv1.getObservable(
+    {did, hostname, env: options.mode});
+
+  // FIXME: move most of this to ext library
+  // remove key
+  //console.log('LEDGER0', JSON.stringify(ledgerDidDocument, null, 2));
+  //console.log('LOCAL0', JSON.stringify(didDocument, null, 2));
+  _removeKeyId(didDocument, options.key);
+  _removeKeyId(ledgerDidDocument, options.key);
+  //console.log('LEDGER1', JSON.stringify(ledgerDidDocument, null, 2));
+  //console.log('LOCAL1', JSON.stringify(didDocument, null, 2));
+
+  // FIXME: reusing import code, rename it
+  _import(options, didDocument);
+
+  const operation = didv1.wrap(
+    {didDocument: ledgerDidDocument, operationType: 'update'});
+  console.log(JSON.stringify(operation, null, 2));
+
+  await release();
+
+  _log(options, 'DID local update successful.');
+
+  //if(options.send) {
+    await _send(options, ledgerDidDocument, didDocument);
```

Comment:
Should be sending `operation` ... or passing `operationType` of `update` into `_send`.